py/TESTING.md

```diff
@@ -0,0 +1,127 @@
+# Python Testing Guide
+
+This guide helps contributors write tests in the Selenium Python codebase.
+
+## Test Framework
+
+* Tests use [pytest](https://pytest.org).
+* Test HTML pages live in `common/src/web/`.
+* `pages` fixture loads test pages via `pages.load("pageName.html")`.
+* Assertions use standard pytest `assert` statements.
+
+```python
+def test_element_is_displayed(driver, pages):
+    pages.load("javascriptPage.html")
+
+    element = driver.find_element(By.ID, "displayed")
+    assert element.is_displayed() is True
+
+@pytest.mark.xfail_safari(reason="Safari doesn't support this")
+def test_something_safari_fails(driver, pages):
+    # Expected to fail on Safari
+    pass
+```
+
+## Test Organization
+
+```
+py/test/
+├── unit/                    # Unit tests (no browser)
+│   └── selenium/webdriver/
+└── selenium/webdriver/      # Integration tests
+    ├── common/              # Cross-browser tests
+    ├── chrome/
+    ├── firefox/
+    ├── safari/
+    └── remote/
+```
+
+Test files end in `_tests.py` (e.g., `visibility_tests.py`).
+
+## Running Tests
```

Comment:
We can also add section to run tests directly with pytest, and the flags we require.

I will add them.